fake_metrics/out/kafkamdm/kafkamdm.go

```diff
@@ -0,0 +1,87 @@
+package kafkamdm
+
+import (
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/raintank/met"
+	"github.com/raintank/raintank-metric/fake_metrics/out"
+	"github.com/raintank/raintank-metric/schema"
+)
+
+type KafkaMdm struct {
+	out.OutStats
+	topic   string
+	brokers []string
+	config  *sarama.Config
+	client  sarama.SyncProducer
+}
+
+func New(topic string, brokers []string, stats met.Backend) (*KafkaMdm, error) {
+	// We are looking for strong consistency semantics.
+	// Because we don't change the flush settings, sarama will try to produce messages
+	// as fast as possible to keep latency low.
+	config := sarama.NewConfig()
+	config.Producer.RequiredAcks = sarama.WaitForAll // Wait for all in-sync replicas to ack the message
+	config.Producer.Retry.Max = 10                   // Retry up to 10 times to produce the message
+	err := config.Validate()
+	if err != nil {
+		return nil, err
+	}
+
+	client, err := sarama.NewSyncProducer(brokers, config)
+	if err != nil {
+		return nil, err
+	}
+
+	return &KafkaMdm{
+		OutStats: out.NewStats(stats, "kafka-mdm"),
+		topic:    topic,
+		brokers:  brokers,
+		config:   config,
+		client:   client,
+	}, nil
+}
+
+func (k *KafkaMdm) Close() error {
+	return k.client.Close()
+}
+
+func (k *KafkaMdm) Flush(metrics []*schema.MetricData) error {
+	preFlush := time.Now()
+	if len(metrics) == 0 {
+		k.FlushDuration.Value(time.Since(preFlush))
+		return nil
+	}
+
+	k.MessageMetrics.Value(1)
+	var data []byte
+
+	for _, metric := range metrics {
+		data, err := metric.MarshalMsg(data[:])
+		if err != nil {
+			return err
+		}
+
+		k.MessageBytes.Value(int64(len(data)))
+
+		prePub := time.Now()
+
+		// We are not setting a message key, which means that all messages will
+		// be distributed randomly over the different partitions.
+		_, _, err = k.client.SendMessage(&sarama.ProducerMessage{
```

Comment:
use `syncProducer.SendMessages(msgs []*ProducerMessage) error`  to send all of the metrics in a batch instead of individually.